pkg/kmm/managedclustermodule.go

```diff
@@ -0,0 +1,301 @@
+
+	return builder
+}
+
+// WithSelector set the selector for the managedclustermodule object.
+func (builder *ManagedClusterModuleBuilder) WithSelector(
+	selector map[string]string) *ManagedClusterModuleBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if selector == nil {
+		builder.errorMsg = "invalid 'selector' argument can not be nil"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.Definition.Spec.Selector = selector
+
+	return builder
+}
+
+// WithOptions creates ManagedClusterModule with generic mutation options.
+func (builder *ManagedClusterModuleBuilder) WithOptions(
+	options ...ManagedClusterModuleAdditionalOptions) *ManagedClusterModuleBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting ManagedClusterModule additional options")
+
+	for _, option := range options {
+		if option != nil {
+			builder, err := option(builder)
+
+			if err != nil {
+				glog.V(100).Infof("Error occurred in mutation function")
+
+				builder.errorMsg = err.Error()
+
+				return builder
+			}
+		}
+	}
+
+	return builder
+}
+
+// PullManagedClusterModule pulls existing module from cluster.
+func PullManagedClusterModule(apiClient *clients.Settings, name, nsname string) (*ManagedClusterModuleBuilder, error) {
+	glog.V(100).Infof("Pulling existing module name %s under namespace %s from cluster", name, nsname)
+
+	builder := ManagedClusterModuleBuilder{
+		apiClient: apiClient,
+		Definition: &mcmV1Beta1.ManagedClusterModule{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the managedclustermodule is empty")
+
+		builder.errorMsg = "managedclustermodule 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the managedclustermodule is empty")
+
+		builder.errorMsg = "managedclustermodule 'namespace' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("managedclustermodule object %s doesn't exist in namespace %s", name, nsname)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Create builds managedclustermodule in the cluster and stores object in struct.
+func (builder *ManagedClusterModuleBuilder) Create() (*ManagedClusterModuleBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating managedclustermodule %s in namespace %s",
+		builder.Definition.Name,
+		builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// Update modifies an existing manageclustermodule on the cluster.
+func (builder *ManagedClusterModuleBuilder) Update() (*ManagedClusterModuleBuilder, error) {
```

Comment:
Does this work correctly without the force argument?

Reply:
I am reusing the same from the KMM Module and it is working as it should. 
Still this specific method did not tested it yet.

Reply:
I think you should at least add this block to the end of the method:
```
if err == nil {
    builder.Object = builder.Definition
}
```
Otherwise, I don't see how builder.Object is being updated after a successful request. Here is an example from another resource that doesn't have a versioned client interface - https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/hive/clusterdeployment.go#L290-292

Reply:
You are right. 

In my case I was updating the object, but I was checking that the operator did the updates. 
And those checks are done with `dmesg`/`lsmod`, for which I did not had to access the `builder.Object `

Updated the code. Thanks !